### DISCUSSION TITLE
Keep getting this error when running make in utilities/nexutil in Raspberry Pi 4b

### DISCUSSION THREAD

**Discussion by Kalinchuk1** [seemoo-lab/nexmon#574] [General]
# make
cd ../libargp && make
make[1]: Entering directory '/home/pi/nexmon/utilities/libargp'
/ndk-build NDK_APPLICATION_MK=`pwd`/Application.mk NDK_APP_OUT=. TARGET_PLATFORM=android-21
/bin/sh: 1: /ndk-build: not found
make[1]: *** [Makefile:2: libs/armeabi/libargp.a] Error 127
make[1]: Leaving directory '/home/pi3/nexmon/utilities/libargp'
make: *** [Makefile:24: ../libargp/local/armeabi/libargp.a] Error 2

Now sure why target_platform is android??

**Comment by jlinktu** [comment]
libargp shoud not be compiled at all when building nexutil on the Raspberry Pi.
I also cannot replicate your problem.
Did you run `source setup_env.sh` in the nexmon root directory before?
The line https://github.com/seemoo-lab/nexmon/blob/76b1edcbd5eb07ef65fd4723be1e75c2734d0fbb/utilities/nexutil/Makefile#L4 should decide if your target is Android or Raspberry Pi. It is based on the output of `uname -m`. What is the output of running `uname -m` on your device?

**Reply by Kalinchuk1** [reply]
I realized what the problem is. Im using kernel 6.1 which is unsupported.

**Reply by jlinktu** [reply]
That is only true for the `brcmfmac` driver.
For me, building nexutil on 6.1 works perfectly fine.

**Comment by Kalinchuk1** [comment]
These are the exact commands I used:
uname -r : 6.1.21-v8+
uname -m: aarch64

sudo su
apt-get update
apt-get upgrade
sudo apt install raspberrypi-kernel-headers git libgmp3-dev gawk qpdf bison flex make autoconf libtool texinfo
git clone https://github.com/seemoo-lab/nexmon.git
cd nexmon
sudo dpkg --add-architecture armhf
sudo apt-get update
sudo apt-get install libc6:armhf libisl23:armhf libmpfr6:armhf libmpc3:armhf libstdc++6:armhf
sudo ln -s /usr/lib/arm-linux-gnueabihf/libisl.so.23.0.0  /usr/lib/arm-linux-gnueabihf/libisl.so.10
sudo ln -s /usr/lib/arm-linux-gnueabihf/libmpfr.so.6.1.0 /usr/lib/arm-linux-gnueabihf/libmpfr.so.4
source setup_env.sh
make
cd patches/bcm43455c0/7_45_206/nexmon
make (After running this, I get this warning: Makefile:59: Warning: Driver for kernel 6.1 not found)
make backup-firmware
make install-firmware (After running this command: Makefile:59: Warning: Driver for kernel 6.1 not found
                                                                                       Makefile:190: Warning: Driver not compiled)
cd nexmon/utilities/nexutil/
make 

and this is when I get this warning:
cd ../libargp && make
make[1]: Entering directory '/home/pi3/nexmon/utilities/libargp'
/ndk-build NDK_APPLICATION_MK=`pwd`/Application.mk NDK_APP_OUT=. TARGET_PLATFORM=android-21
/bin/sh: 1: /ndk-build: not found
make[1]: *** [Makefile:2: libs/armeabi/libargp.a] Error 127
make[1]: Leaving directory '/home/pi3/nexmon/utilities/libargp'
make: *** [Makefile:24: ../libargp/local/armeabi/libargp.a] Error 2
